spec/factories/payments.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :payment do
+    paid_on { Faker::Date.between(from: care_finished_on.strftime('%Y-%m-%d'), to: "#{Time.zone.today.strftime('%Y-%m-%d')}") }
+    care_started_on { Faker::Date.backward(365).strftime('%Y-%m-%d') }
+    care_finished_on { Faker::Date.between(from: care_started_on.strftime('%Y-%m-%d'), to: "#{Time.zone.today.strftime('%Y-%m-%d')}").strftime('%Y-%m-%d') }
+    amount { Faker::Number.between(from: 0, to: 99_999_999) }
+    discrepancy { Faker::Number.between(from: 0, to: 999_999) }
```

Comment:
For accuracy's sake, I'd love to save the amount to a variable and then get a discrepancy that's less than that amount.

i.e.

```
FactoryBot.define do
  payment_amount = Faker::Number.between(from: 1, to: 99_999_999)
  factory :payment do
    ...
    amount { payment_amount }
    discrepancy { Faker::Number.between(from: 0, to: payment_amount) }
```

Reply:
Don't have to even save it to a variable.  You can define amounts based on other attributes.  
ex: `discrepancy { Faker::Number.between(from: -0, to: amount) }`
(which is how I have it defined ATM.)